nansat/mappers/mapper_metno_hfr.py

```diff
@@ -0,0 +1,134 @@
+                continue
+            gdal_subdataset = gdal.Open(subdataset[0])
+            # need to be float for the nan replasement
+            band_data = gdal_subdataset.GetRasterBand(int(src_timestamp_id)).ReadAsArray().astype('float')
+            # remove fill value (replace with nan)
+            fill_value = int(gdal_subdataset.GetMetadata_Dict()['#'.join([subdataset_name, '_FillValue'])])
+            band_data[band_data == fill_value] = np.nan
+            # Interpolate data on the regular grid
+            band_grid_data = self.band2grid((nc_dataset['x'][:], nc_dataset['y'][:]), 
+                                             band_data, (x_grd, y_grd))
+            # Create VRT ffor the regridded data
+            band_vrt = VRT.from_array(band_grid_data)
+            # Add VRT to the list of all dataset vrts
+            self.band_vrts[subdataset_name + 'VRT'] = band_vrt
+            # Add band to the dataset
+            src = {'SourceFilename': self.band_vrts[subdataset_name + 'VRT'].filename, 
+                   'SourceBand': 1}
+            # Add band specific metadata
+            dst = {'name': subdataset_name}
+            for key in gdal_subdataset.GetMetadata_Dict().keys():
+                if key.startswith(subdataset_name):
+                    clean_metadata_name = key.split('#')[1]
+                    dst[clean_metadata_name] = gdal_subdataset.GetMetadata_Dict()[key]
+            # Create band
+            self.create_band(src, dst)
+            self.dataset.FlushCache()
+
+        # Set GCMD metadata
+        self.dataset.SetMetadataItem('instrument', json.dumps(pti.get_gcmd_instrument('SCR-HF')))
+        self.dataset.SetMetadataItem('platform', json.dumps(pti.get_gcmd_platform('CODAR SeaSonde')))
+        self.dataset.SetMetadataItem('Data Center', json.dumps(pti.get_gcmd_provider('NO/MET')))
+        self.dataset.SetMetadataItem('Entry Title', 'Near-Real Time Surface Ocean Radial Velocity')
+        self.dataset.SetMetadataItem('gcmd_location',json.dumps(pti.get_gcmd_location('NORTH SEA')))
+        # Set time coverage metadata
+        self.dataset.SetMetadataItem('time_coverage_start', timestamp.isoformat())
+        self.dataset.SetMetadataItem('time_coverage_end',
+                                     (timestamp + timedelta(minutes=59, seconds=59)).isoformat())
+        # Set NetCDF dataset metadata
+        for key, value in gdal_dataset.GetMetadata_Dict().items():
+            self.dataset.SetMetadataItem(key.split('#')[1], value)
+
+    def date_from_filename(self, src):
+        filename = os.path.split(src)[-1].split('.')[0]
+        year, month, day = filename.split('_')[-3:]
+        src_timestamp = datetime(int(year), int(month), int(day), 0, 0, 0)
+        return src_timestamp
+
+    def create_linear_grid(self, x, y, px_size):
+        x_grd, y_grd = np.meshgrid(np.arange(x.min(), x.max(), px_size),
+                                   np.arange(y.max(), y.min(), px_size * -1))
```

Comment:
I'm a bit worried about that. in this solution x.max() will not be in x_grd. Because `np.arange(1,10,1)` returns `array([1, 2, 3, 4, 5, 6, 7, 8, 9])`. Maybe you need `np.arange(x.min(), x.max()+px_size, px_size)` ?